### PR TITLE
Add stratum-bmv2 Docker build script

### PR DESCRIPTION
**Pull request by bocon13**
Update the release script to use the build script

### CODE REVIEW

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+set -x
+docker run --rm \
+  $DOCKER_OPTS \
+  $DOCKER_EXTRA_RUN_OPTS \
+  -v $STRATUM_ROOT:/stratum \
+  -v $(pwd):/output \
+  -w /stratum \
+  --entrypoint bash \
+  $DOCKER_IMG -c \
+    "bazel build //stratum/hal/bin/bmv2:stratum_bmv2_deb.deb \
```
Comment (pudelkoM) [inline tools/mininet/build-stratum-bmv2-container.sh:64]:
```suggestion
    "bazel build //stratum/hal/bin/bmv2:stratum_bmv2_deb \
```

File suffix is not really necessary.

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+     cp -f /stratum/bazel-bin/stratum/hal/bin/bmv2/stratum_bmv2_deb.deb /output/"
+set +x
+
+
+DOCKER_BUILD_OPTS=""
+if [ "$(docker version -f '{{.Server.Experimental}}')" = "true" ]; then
+  DOCKER_BUILD_OPTS+="--squash "
+fi
+
+DOCKER_BUILD_OPTS+="--label build-timestamp=$(date +%FT%T%z) "
```
Comment (pudelkoM) [inline tools/mininet/build-stratum-bmv2-container.sh:76]:
```suggestion
DOCKER_BUILD_OPTS+="--label stratum-target=$STRATUM_TARGET "
DOCKER_BUILD_OPTS+="--label build-timestamp=$(date +%FT%T%z) "
```

Both BCM and BF containers have that label.

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# Copyright 2018-present Open Networking Foundation
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+DOCKERFILE_DIR=$( cd $(dirname "${BASH_SOURCE[0]}") >/dev/null 2>&1 && pwd )
+STRATUM_ROOT=${STRATUM_ROOT:-"$( cd "$DOCKERFILE_DIR/../.." >/dev/null 2>&1 && pwd )"}
+JOBS=${JOBS:-4}
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:8]:
```suggestion
JOBS=${JOBS:-4}
STRATUM_TARGET=stratum-bmv2
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  GIT_REF=$(git describe --tags --no-match --always --abbrev=40 --dirty | sed -E 's/^.*-g([0-9a-f]{40}-?.*)$/\1/')
+  GIT_SHA=$(git describe --tags --match XXXXXXX --always --abbrev=40 --dirty)
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.source=$GIT_URL "
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.version=$GIT_REF "
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.revision=$GIT_SHA "
+fi
+popd
+
+# Build Stratum BF runtime Docker image
+STRATUM_NAME=stratum-bmv2
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:92]:
```suggestion
STRATUM_NAME=$STRATUM_TARGET
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+RUNTIME_IMAGE=opennetworking/mn-stratum
+echo "Building Stratum runtime image: $RUNTIME_IMAGE"
+set -x
+docker build \
+  $DOCKER_BUILD_OPTS \
+  -t "$RUNTIME_IMAGE" \
+  -f "$DOCKERFILE_DIR/Dockerfile" \
+  "$(pwd)"
+
+docker save $RUNTIME_IMAGE | gzip > ${STRATUM_NAME}-${SDE_VERSION}-docker.tar.gz
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:104]:
```suggestion
docker save $RUNTIME_IMAGE | gzip > ${STRATUM_NAME}-docker.tar.gz
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  GIT_REF=$(git describe --tags --no-match --always --abbrev=40 --dirty | sed -E 's/^.*-g([0-9a-f]{40}-?.*)$/\1/')
+  GIT_SHA=$(git describe --tags --match XXXXXXX --always --abbrev=40 --dirty)
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.source=$GIT_URL "
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.version=$GIT_REF "
+  DOCKER_BUILD_OPTS+="--label org.opencontainers.image.revision=$GIT_SHA "
+fi
+popd
+
+# Build Stratum BF runtime Docker image
+STRATUM_NAME=$STRATUM_TARGET
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:94]:
```suggestion
STRATUM_NAME=$(echo $STRATUM_TARGET | sed 's/_/-/')
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 7]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Copyright 2018-present Open Networking Foundation
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+DOCKERFILE_DIR=$( cd $(dirname "${BASH_SOURCE[0]}") >/dev/null 2>&1 && pwd )
+STRATUM_ROOT=${STRATUM_ROOT:-"$( cd "$DOCKERFILE_DIR/../.." >/dev/null 2>&1 && pwd )"}
+JOBS=${JOBS:-4}
+STRATUM_TARGET=stratum-bmv2
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:9]:
```suggestion
STRATUM_TARGET=stratum_bmv2
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+set -x
+docker run --rm \
+  $DOCKER_OPTS \
+  $DOCKER_EXTRA_RUN_OPTS \
+  -v $STRATUM_ROOT:/stratum \
+  -v $(pwd):/output \
+  -w /stratum \
+  --entrypoint bash \
+  $DOCKER_IMG -c \
+    "bazel build //stratum/hal/bin/bmv2:stratum_bmv2_deb \
```
Comment (bocon13) [inline tools/mininet/build-stratum-bmv2-container.sh:65]:
```suggestion
    "bazel build //stratum/hal/bin/bmv2:${STRATUM_TARGET}_deb \
```

File: tools/mininet/build-stratum-bmv2-container.sh [hunk 9]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Copyright 2018-present Open Networking Foundation
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+DOCKERFILE_DIR=$( cd $(dirname "${BASH_SOURCE[0]}") >/dev/null 2>&1 && pwd )
+STRATUM_ROOT=${STRATUM_ROOT:-"$( cd "$DOCKERFILE_DIR/../.." >/dev/null 2>&1 && pwd )"}
+JOBS=${JOBS:-4}
+STRATUM_TARGET=stratum_bmv2
```
Comment (pudelkoM) [inline tools/mininet/build-stratum-bmv2-container.sh:9]:
```suggestion
STRATUM_TARGET=stratum_bmv2
JOBS=${JOBS:-4}
```

**Review by pudelkoM — APPROVED** [review]

**Pull request merged**